articles/job-scheduling-in-node.js/index.md

```diff
@@ -0,0 +1,146 @@
+1. Have Node.js installed.
+2. Have a basic knowledge of Node.js. [Here](/engineering-education/history-of-nodejs/) is an article to get you started.
+3. Have a basic knowledge of the `Express framework`. [Here](/engineering-education/express/) is an article to get you started.
+4. Be familiar with sending emails through `nodemailer`. Refer to [this](/engineering-education/node-mailer/) article for a deep dive into node-mailer.
+
+Let's get started.
+
+### Step 1 — Creating a Node Application
+Create a directory and give it a name of your choice. I'll name mine (node-cron-example). Then open the directory you have created using your favorite code editor. Now open the terminal and type
+
+```bash
+npm init
+```
+Fill the required fields to create a package.json file that will be used to track dependencies. Add the express framework, node-cron module and node-mailer module by typing
+
+```bash
+npm install express node-cron nodemailer
+```
+
+### Step 2 — Creating The Server
+Now that we have created our node application and initialized it. Let's go ahead and create the server that we will use for scheduling tasks. First, type the code below to import express module and initialize it
+
+```Javascript
+const express = require('express');
+const app = express();
+```
+
+Next up, set the server to listen to port 2400. **Note: You can use a port number of your choice**
+
+```Javascript
+app.listen(2400, () => {console.log("Server started at port 2400")});
+```
+
+That's all we need for our server now. To start the server type 'node <entry point file>' on the terminal as shown
+
+```bash
+node index
+```
+
+This is the expected output from the console
+```
+Server started at port 2400
+```
+
+### Step 3 — Scheduling a Task
+Now that we have our server up and running. Let's see how to schedule a simple task before we get into sending our emails. Let's start by importing the node-cron module
+
+```Javascript
+const cron = require('node-cron');
+```
+
+To schedule a task, we pass in our *crontab* expression and the function we want to schedule on a schedule method. In this example, we  pass the default `cron` expression which will schedule the task to run every minute.
+
+```Javascript
+cron.schedule('* * * * *', () => {console.log("Task is running every minute")});
+```
+The console output is as shown
+
+```bash
+$ node index
+Server started at port 2400
+Task is running every minute
+Task is running every minute
+Task is running every minute
+...
+```
+### Step 4 — Creating The Email Function
+Let's create the function that will send emails.  First, import the node-mailer module
+
+```Javascript
+const mailer = require('nodemailer');
+```
+
+Then create a function and give it a name of your choice.
+In the function, we will create a transporter through nodemailer's `createTransport` method. We will then use the `sendMail` method to send the email.
+
+```Javascript
+function sendEmail(message){
+    // Creating a transporter
```

Comment:
You can initialize the `transporter` outside the function so that the transporter can seem reusable and cleaner code. Since the `sendEmail` is called several times, it won't be efficient to recreate the `transporter` every time.